libdenaro/src/helpers/currencyhelpers.cpp

```diff
@@ -44,24 +44,36 @@ namespace Nickvision::Money::Shared
         if(!systemCurrency)
         {
             systemCurrency = std::make_unique<Currency>();
-            systemCurrency->setSymbol(std::use_facet<std::moneypunct<char>>(std::locale("")).curr_symbol());
-            systemCurrency->setCode(std::use_facet<std::moneypunct<char, true>>(std::locale("")).curr_symbol());
-            systemCurrency->setDecimalSeparator(std::use_facet<std::moneypunct<char>>(std::locale("")).decimal_point());
-            systemCurrency->setGroupSeparator(std::use_facet<std::moneypunct<char>>(std::locale("")).thousands_sep());
+            std::locale locale;
+            try
+            {
+                locale = std::locale("");
+            }
+            catch (...)
+            {
+                fprintf(stderr, "Unknown locale on system. Defaulting to USD as the currency");
```

Comment:
You can make this C++ instead of C:
```cpp
std::err << "Unknown system locale. Reverting to USD currency." << std::endl;
```

Just be sure to add `#include <iostream>` after `#include <iomanip>`.